File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n=height.length-1;
        int left=0,right=n,maxarea=0;
        while(left<right){
            int currheight=Math.min(height[left],height[right]);
            int width=right-left;
            int area=currheight*width;
            maxarea=Math.max(area,maxarea);
            if(height[left]<height[right]){
                left++;
            }
            else{
                right--;
            }
        }
        return maxarea;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left,right,maxarea=0,len(height)-1,0
        while left<right:
            currheight=min(height[left],height[right])
            width=right-left
            area=currheight*width
            maxarea=max(area,maxarea)
            if height[left]<height[right]:
                left+=1
            else:
                right-=1
        return maxarea
        